Equilobe.DailyReport.BL/Jira/SprintLoader.cs

```diff
@@ -22,64 +22,68 @@ public SprintLoader(ProjectDateFilter filter, JiraClient client)
             Client = client;
         }
 
-        List<View> GetRapidViewsFromProject()
+        public SprintContext GetSprintDetails()
         {
-            var views = Client.GetRapidViews();
+            var boardId = Client.Board(Filter.ProjectKey).Id;
 
-            var rapidViews = views
-                .Where(view => view.filter.name.ToLower().Contains(Filter.ProjectKey.ToLower())
-                            || view.filter.name.ToLower().Contains(Filter.ProjectName.ToLower()))
-                .Select(view => view)
-                .ToList();
+            if (string.IsNullOrEmpty(boardId))
+                return null;
 
-            return rapidViews;
-        }
+            var lastSprints = GetLastSprints(boardId);
 
-        View GetActiveView(List<View> views)
-        {
-            return views.Find(v => v.sprintSupportEnabled == true);
-        }
-
-        RapidView GetRapidView(string activeViewId)
-        {
-            return Client.GetRapidView(activeViewId);
+            return GetSprintContext(lastSprints);
         }
 
-        public SprintContext GetSprintDetails()
+        private List<Sprint> GetLastSprints(string boardId)
         {
-            var rapidViewId = GetRapidViewId();
-            if (rapidViewId == null)
-                return null;
+            var startAt = (long)0;
+            var total = (long)0;
 
-            var sprints = Client.GetAllSprints(rapidViewId, Filter.ProjectKey);
+            while (true)
+            {
+                var sprintsResponse = Client.GetAllSprints(boardId, startAt.ToString());
+                
+                if (sprintsResponse.Values.Capacity < 3)
+                {
+                    if (total == 0)
+                        return sprintsResponse.Values;
 
-            if (sprints == null || sprints.Count == 0)
-                return null;
+                    if (sprintsResponse.IsLast)
```

Comment:
Why do you verify this twice?

Reply:
The first case scenario for having the last sprints on the response is that we have less than 3 and it's not the first request (total is not 0), which means that it's needed to make another request with the startingPoint a few items before so that we'll have 3 sprints

The second case scenario is that we have more than 3 sprints on the response, and that's just ok.